### PR TITLE
feat: add args, kwargs to task

### PR DESCRIPTION
**Pull request by kennhh**
## Pull Request Type

- [x] Feature addition
- [ ] Bugfix
- [ ] Documentation update
- [ ] Code refactor
- [ ] Tests improvement
- [ ] CI/CD pipeline enhancement
- [ ] Other: [Replace with a description]

## Description

Adds variable length argument list and keyword arguments to task.
Example:
```py
@Task.create(...)
async def task(*args, **kwargs):
     ...

...
task.start(*args, **kwargs)
```

## Changes

- added args, kwargs to task

## Python Compatibility

- [ ] I've ensured my code works on Python `3.10.x`
- [x] I've ensured my code works on Python `3.11.x`

## Checklist

- [x] I've run the `pre-commit` code linter over all edited files
- [ ] I've tested my changes on supported Python versions
- [ ] I've added tests for my code, if applicable
- [ ] I've updated / added  documentation, where applicable

### CODE REVIEW

File: interactions/models/internal/tasks/task.py [hunk 1]
```diff
@@ -75,25 +75,25 @@ def on_error(self, error: Exception) -> None:
...
 
             if isinstance(val, BaseTrigger):
                 self.reschedule(val)
         except Exception as e:
             self.on_error(e)
 
-    def _fire(self, fire_time: datetime) -> None:
+    def _fire(self, fire_time: datetime, *args, **kwargs) -> None:
         """Called when the task is being fired."""
-        self.trigger.set_last_call_time(fire_time)
```
Comment (AstreaTSS) [inline interactions/models/internal/tasks/task.py:92]: Why is this changed?

**Review by AstreaTSS — CHANGES_REQUESTED** [review]

File: interactions/models/internal/tasks/task.py (hunk 1 above)
Comment (kennhh) [reply interactions/models/internal/tasks/task.py:92]: i just couldn't traceback to set_last_call_time. might be an issue on my end but the quickest fix i found was just to use last_call_time

Comment (AstreaTSS) [reply interactions/models/internal/tasks/task.py:92]: Suggesting this was added in another PR to fix something (#1438), I'm not sure if this is really the right solution.

Comment (kennhh) [reply interactions/models/internal/tasks/task.py:92]: yeah, you're definitely right. seems i missed that pr (oops) so ill change it back. thanks for telling me!

**Review by AstreaTSS — APPROVED** [review]

**Review by i0bs — APPROVED** [review]
